packages/url_launcher/url_launcher_android/android/src/main/java/io/flutter/plugins/urllauncher/UrlLauncher.java

```diff
@@ -118,6 +156,87 @@ public void closeWebView() {
     applicationContext.sendBroadcast(new Intent(WebViewActivity.ACTION_CLOSE));
   }
 
+  @TargetApi(Build.VERSION_CODES.R)
+  private static @NonNull Boolean launchNativeApi30(@NonNull Context context, @NonNull Uri uri) {
+    Intent nativeAppIntent =
+        new Intent(Intent.ACTION_VIEW, uri)
+            .addCategory(Intent.CATEGORY_BROWSABLE)
+            .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_REQUIRE_NON_BROWSER);
```

Comment:
This part is useful functionality, but should be controlled by the launch mode — this is implementing `LaunchMode.externalNonBrowserApplication`.

An implementation of that would resolve https://github.com/flutter/flutter/issues/66721 and would also be quite welcome. It's probably simplest to handle that as a separate PR, though. In order to make it properly conditional on the launch mode, I think it may also require changes in `url_launcher_platform_interface`.